### ISSUE TITLE
& + & isn't working 

### ISSUE THREAD

**Issue by it1011** [emotion-js/emotion#1922]
**Current behavior:**

I made a tab components using @emotion/styled, I want each tab margin left 20px except the first one. so I used & + & to refer each tab compnent I want to style. 

**To reproduce:**

I made a example here ==>  https://stackblitz.com/edit/react-pcrmlh

found that when there are some logic inside the css delcaration,  classname changed,  but the css rules does not response to that. 

![image](https://user-images.githubusercontent.com/30334815/85963682-71717180-b9e9-11ea-9a98-2d41dab49b61.png)
![image](https://user-images.githubusercontent.com/30334815/85963698-7fbf8d80-b9e9-11ea-99b3-cb2a7d8c201e.png)

**Expected behavior:**

& + & should be able to refer to self correctly

**Environment information:**

![image](https://user-images.githubusercontent.com/30334815/85963828-ff4d5c80-b9e9-11ea-8052-14b9b7f7f7ef.png)

**Comment by ambertino** [comment]
`&` is equal to the class name generated for each set of properties. So the class name for your active item is different to the inactive ones. You can check them my inspecting the DOM. Therefore the CSS rule is not applied.

**Comment by ambertino** [comment]
There is already related issue: https://github.com/emotion-js/emotion/issues/1827

**Comment by it1011** [comment]
> that

I understant  the solution, but I don't think  :not(:first-child)  will work because the logic inside cause generating new css rules and new classname which make the  css rules which rely on sibling elements' relationship broken in this case. 

I tested styled-components for the same case, it seems generating extra class name to maintain the  relationship instead of breaking it.

**Comment by Andarist** [comment]
> I tested styled-components for the same case, it seems generating extra class name to maintain the relationship instead of breaking it.

Yes, they introduce extra class names to accommodate for that - we do not plan to do that though. Especially given that the preferred (by our team) way to style components is through the `css` prop and in this case it's just impossible to do this as styles are not owned by any particular component - there are just styles that you can apply to some components but there is no direct relationship between those 2.

**Comment by ambertino** [comment]
> I understant the solution, but I don't think :not(:first-child) will work because the logic inside cause generating new css rules and new classname which make the css rules which rely on sibling elements' relationship broken in this case.
It should work. Check this out:
https://stackblitz.com/edit/react-j5jnjy

**Comment by it1011** [comment]
> > I understant the solution, but I don't think :not(:first-child) will work because the logic inside cause generating new css rules and new classname which make the css rules which rely on sibling elements' relationship broken in this case.
> 
> It should work. Check this out:
> https://stackblitz.com/edit/react-j5jnjy

🙈, You're right, I think I  had misunderstanding of this pseudo-class, I didn't test the solution  myself,  thanks a lot for correcting me。😄

**Comment by it1011** [comment]
> > I tested styled-components for the same case, it seems generating extra class name to maintain the relationship instead of breaking it.
> 
> Yes, they introduce extra class names to accommodate for that - we do not plan to do that though. Especially given that the preferred (by our team) way to style components is through the `css` prop and in this case it's just impossible to do this as styles are not owned by any particular component - there are just styles that you can apply to some components but there is no direct relationship between those 2.

Mm.., I think I got your point, but I still have question,  shouldn't relationship be part of styling, how relationship be represented from emotion's solution? I went throuth the docs, seem there is no clearification for that。

**Comment by Andarist** [comment]
Creating those relationships in CSS makes styling **less** predictable and also might break our composition patterns. When we write this:
```js
[fooStyles, { color: 'hotpink' }]
```
the expectation is that the color will be `'hotpink'`, but if you introduce any combinators to `fooStyles` or in other styles this doesn't hold true anymore, because the CSS specificity wins. For example, in React Native there are no combinators like in the CSS and people still manage to style their apps, it requires some shift of the mind, but it actually comes with good benefits. Some existing CSS design systems/styling approaches also ban those combinators, so it's not like we are alone in this.

**Comment by divmgl** [comment]
I'm running into this now too. My workaround is that I've created wrappers in order to get the effect that I want. The wrapper's styles are always static, and the descendants have the dynamic styles. It's not ideal, and I *was* expecting '& + &' to work. A suggestion would be to create a new hash when encountering '&' styles.

**Comment by it1011** [comment]
> Creating those relationships in CSS makes styling **less** predictable and also might break our composition patterns. When we write this:

I agree with you,  should this be the core concept of emotion？and worth being put into the documentation
if there were some code of conduct for how to solve the problem like & + & will be much better。

I having been using styled-components and  attracted by the smaller size of emotion, want to give it a try. So I expected the same behavior as styled-components when I am using @emotion/styled. 

hight performance、modular css、logic allowed and full styling capability  is the why I choose a css in js library, I really want it to be "perfect"😂

**Comment by Andarist** [comment]
I agree this could be documented - documentation PRs are always welcome.
